src/common/calculation.js

```diff
@@ -1,25 +1,28 @@
-import axios from 'axios'
 import BN from 'bignumber.js'
+import EthDater from 'ethereum-block-by-date'
 import { getXVaderPriceByBlock } from './graphql'
 import { getStartOfTheDayTimeStamp, getCompoundApy } from './utils'
 import defaults from './defaults'
 
 const getXVaderPrice = () => getXVaderPriceByBlock()
 
 const getBlockNumberPriorDaysAgo = async (numberOfDays) => {
+	const dater = new EthDater(defaults.network.provider)
 	const SECOND_IN_A_DAY = 86400
 	const startOfDayTs = getStartOfTheDayTimeStamp()
 	const sevenDaysAgoTs = startOfDayTs - SECOND_IN_A_DAY * numberOfDays
-	const url = `${defaults.api.etherscanApiUrl}/api
-		?module=block
-		&action=getblocknobytime
-		&timestamp=${sevenDaysAgoTs}
-		&closest=before
-		&apikey=${defaults.api.etherscanApiKey}
-	`
+	const timestampInMs = sevenDaysAgoTs * 1000
+	const cachedHits = localStorage.getItem('BLOCK_NUMBER_BY_TIMESTAMP')
```

Comment:
it might take several requests to block chain to get the data so I cache it into local storage